opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -2290,6 +2290,73 @@ namespace detail {
 
 
 
+
+    template <class Grid, class WellModel, class Implementation>
+    std::vector<V>
+    BlackoilModelBase<Grid, WellModel, Implementation>::
+    computeFluidInPlace(const ReservoirState& x,
+                        const WellState& xw,
+                        const std::vector<int>& fipnum)
+    {
+        using namespace Opm::AutoDiffGrid;
+        const int nc = numCells(grid_);
+        const int np = x.numPhases();
+
+        SolutionState state(np);
```

Comment:
>If you can do with the accum data, why was it not used here?

Because accum add rs and rv to oil and gas term, they're need to be separated for FIP.

>why not accum[1]

Actually, OOIP  need accum[0] and currecntly FIP need accum[1]